### DISCUSSION TITLE
Documentation : Auto generate elements tables

### DISCUSSION THREAD

**Discussion by arnaudroubinet** [structurizr/java#323] [Q&A]
### Description

Most of the time in our documentation, we want to list elements we describe with a table format.
Unless I've missed something, today I have to describe the elements twice, in the documentation and in the dsl file.

To illustrate my point, here's an example :
```
        my_software = softwareSystem "My super software with a wonderful and explicit name"{
                super_container = container "A super container"{
                    technology "A wonderful one"
                    tags tag_1 tag_2
                }
                super_container2 = container "Another super container"{
                    technology "A wonderful one"
                    tags tag_2 tag_3
                }
               ...
        }
```

I would like structurizr to generate the following markdown.

| Container                | technology      | tags        |
| ------------------------ | --------------- | ----------- | 
| A super container        | A wonderful one | tag_1 tag_2 |
| Another  super container | A wonderful one | tag_2 tag_3 |
...

Of course, it would be perfect if we could define the columns to be displayed and potentially the order, filter on tags, etc.

In this way, we only need to describe the element once for the entire documentation.

### Priority

I have no budget and there's no rush, please add this feature for free

**Comment by arnaudroubinet** [comment]
Probably not the good repository ?

**Comment by simonbrowndotje** [comment]
I'd recommend writing a DSL plugin to do this, which should be relatively straightforward and give you complete control over how everything is rendered.

1. Add a marker into your Markdown (e.g. `${ELEMENTS_TABLE}`).
2. Write a plugin that reads all documentation sections and replaces the above marker with a table.

Here's an example to get you started -> https://github.com/structurizr/examples/tree/main/dsl/plantuml-and-mermaid/plugin/src/main/java/plantuml

**Comment by arnaudroubinet** [comment]
Thanks for these elements.

If I find the time to write plugins, how do you manage sharing them?

Do I make a PR in this repository to make it accessible to everyone? Or do you plan for each person to manage their own plugin and add it to their instance if they wish?

**Reply by simonbrowndotje** [reply]
If you'd like to share your plugin with the community, the best option is to publish it on your own GitHub repo.
